remote-query/remote-query-server/src/main/java/org/infinispan/query/remote/impl/ObjectRemoteQueryEngine.java

```diff
@@ -17,7 +17,7 @@
  */
 class ObjectRemoteQueryEngine extends QueryEngine<Descriptor> {
 
-   private final EmbeddedQueryFactory queryFactory = new EmbeddedQueryFactory(this);
+   private final EmbeddedQueryFactory queryFactory = new EmbeddedQueryFactory(this, false);
```

Comment:
shouldn't check `cache.hasFlag(CACHE_MODE_LOCAL)` instead of `false`?

Reply:
Yes, this class is only used to query from remote clients, I ignored them for now as it's harder to target individual nodes from those clients

Reply:
I will take a look nonetheless